### PR TITLE
feat(i18n): Upgrade i18next

### PR DESCRIPTION
**Pull request by annilsson**
**What is the problem this PR is trying to solve?**
Using the react-i18next `translate` HOC comes with some downsides, like:

- In unit tests if you import the default exported component you're not testing _your_ component, but the i18next component (HOC). Snapshots become huge with i18next object attached. You can get around it by importing the component without the HOC, or using `MyComponent.WrappedComponent`, but it's easy to forget because not all components use the HOC.
- When using the `t` function you need to set defaultProp or it will throw errors in unit tests
- Component displaynames will become things like `Translate(MyComponent)` often adding confusion. 

**What is the chosen solution to this problem?**
Upgrade i18next and react-i18next so we can use the `useTranslation` hook ([https://react.i18next.com/latest/usetranslation-hook](https://react.i18next.com/latest/usetranslation-hook)). The exported component and its displaynames remains the same always, and `t` is no longer a prop, so no need for `defaultProps`. 

You don't **have** to use the hook if you don't want to. You can keep using the HOC, _but it has been renamed_ from `translate` to `withTranslation`. 

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**
There is a codemod to change `translate()` import and use
```bash
npx jcodeshift <talend_ui_path>/codemodes/i18next/from_9_to_15.js <your_app_src>
```

You will likely need to update your tests
* snapshots: displayName has changed : `Translate(MyComponent)` --> `withI18nextTranslation (MyComponent)` and inserted `<I18n>` component has been removed from component tree.
* selectors: if you selected `Translate(MyComponent)` you now need to select `withI18nextTranslation (MyComponent)`

If you attach sub components to your main components (like `List.Item` for example) you need to pay attention to how you export them:

https://react.i18next.com/latest/withtranslation-hoc#hoist-non-react-statics
```js
// Wrong, this will result in having to import `MyComponent.WrappedComponent.Item`
MyComponent.Item = SomeOtherComponent;
export default withTranslation()(MyComponent);

// Correct, here you can import `MyComponent.Item`
const TranslatedMyComponent = withTranslation()(MyComponent);
TranslatedMyComponent.Item = SomeOtherComponent;
export default TranslatedMyComponent;
```

### CODE REVIEW

File: codemodes/i18next/codemod.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	return foundAndReplaced ? result : null;
+}
+
+function replaceTranslateCalls(file, j) {
+	return j(file)
+		.find(j.CallExpression, { callee: { name: 'translate' } })
+		.forEach(nodePath => {
+			nodePath.value.callee = j.identifier('withTranslation');
+		})
+		.toSource();
```
Comment (jmfrancois) [inline codemodes/i18next/codemod.js]: In other code mode we call toSource with the following options:
```
toSource({ useTabs: true, quote: 'single', trailingComma: true })
```

why this is not the case here ?

File: codemodes/i18next/codemod.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+export default function transformer(file, api) {
+	/*
+		return j(file.source)
+			.find(j.Identifier)
+			.filter(nodePath => nodePath.node.name === 'translate')
+			.forEach(nodePath => {
+				j(nodePath).replaceWith(j.identifier("withTranslation"));
+		  })
+			.toSource();
+		  */
```
Comment (jmfrancois) [inline codemodes/i18next/codemod.js]: dead code please remove

File: codemodes/i18next/codemod.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-param-reassign */
```
Comment (jmfrancois) [inline codemodes/i18next/codemod.js]: please rename this file in sth more scriptish : `from_nine_to_fifteen.js` ?

File: codemodes/i18next/codemod.js (hunk 1 above)
Comment (jsomsanith-tlnd) [reply codemodes/i18next/codemod.js]: Oh yes thanks !

File: codemodes/i18next/codemod.js (hunk 2 above)
Comment (jsomsanith-tlnd) [reply codemodes/i18next/codemod.js]: Yep, just saw it / removed it :)

**Pull request merged**